### ISSUE TITLE
Firebase Invites never survive App Installation Process on iOS

### ISSUE THREAD

**Issue by davidwernhart** [firebase/firebase-ios-sdk#1403]
* Xcode version: 9.4
  * Firebase SDK version: 5.2.0
  * Firebase Component: Invites
  * Component version: 3.0.0

Steps to reproduce:

* Download Firebase Invites example project:
https://github.com/firebase/quickstart-ios/tree/master/invites

* Run Pod Install

* Download GoogleService-Info.plist, enter the Reversed_Client_Id in the URL Types and the Associated Domains from your Firebase Project

* Compile and run the app

* Send an Invite through the app

* Uninstall the app, click on invite Link, you get to the App Store page

* Install the app through Xcode, the link the app is receiving from ApplicationOpenURLOptions is always just "myAppScheme://google/link/?dismiss=1&is_weak_match=1"

I am trying to implement Firebase Invites on iOS right now, and after an app install I am also always receiving "myAppScheme://google/link/?dismiss=1&is_weak_match=1" instead of the deep link. Therefore FIRInvites cannot parse it and it does not work. Is this related to the issue with Dynamic Links? I have one iPhone 6 on iOS 11.4 and one iPhone 5 on 10.3.3 and it does not work on both of them.

Since I was not sure if there was an error in my implementation of the Invites SDK, i downloaded the quickstart Invite project from Github and it did not work there eighter. My iOS App in the Firebase Console has App - ID, Bundle - ID, App Store ID and Team ID all filled in, and Dynamic Links are activated.

Mind you that it does work on android (with the same firebase project) and on iOS when the app is already installed, so this problem is just occuring when the user is first installing the app after receiving an Invite.

**Comment by davidwernhart** [comment]
any updates on this one?

**Comment by morganchen12** [comment]
Does this occur on all iOS versions after #1244 is fixed?

**Comment by davidwernhart** [comment]
It is at least on 11.4

Are you able to reproduce it? Maybe the issue is on my end after all.

**Comment by nikeshakya** [comment]
@morganchen12 I am experiencing same issue..
invite url works only when app is in background mode...
fresh install and terminated app state doesn't respond to dynamic link
Any update on this?

**Comment by morganchen12** [comment]
No updates yet, but it does look like this is a bug. Mandar, can you take a look when you get a chance?

**Comment by nikeshakya** [comment]
@morganchen12 thank you for prompt reply.
I hope this gets fixed asap.

**Comment by Maikol** [comment]
@morganchen12 I'm having the same issue.
Any update? Thank you

**Comment by nikeshakya** [comment]
@dmandar @morganchen12 please fix this issue asap...we need workable invites in our app

**Comment by touqeershafi** [comment]
we have build our whole feature of refer a friend on this firebase dynamic link and suddenly you came to know that firebase dynamic link is not working :(

**Comment by nathankallman** [comment]
We are also still experiencing the same issue on fresh iOS installs with Firebase Dynamic links (SDK version: 5.5.0) Android and post-install iOS are working as expected.

Is there an update to this issue on what we should expect?

**Comment by hugocroch** [comment]
@nathankallman. It seems that I have the same issue. Also, I cannot reproduce the bug on Android.

**Comment by PeeGee85Spil** [comment]
Having the same problem with FIrebase Dynamic Links on iOS. Normal dynamic links (when the app is already installed) work fine, deferred dynamic links lead to the store, I install the app and open it, nothing (application:openURL:sourceApplication:annotation: is not called). If I then close and restart the app, application:openURL:sourceApplication:annotation: is called and I do get the deeplink. I'm 99.9% sure that this used to work properly say half a year ago.

By the way, if I click the link, go to the store, then install the app from XCode, the deeplink works fine. With a build on the App Store it fails though.

Here's my link by the way: https://pu68v.app.goo.gl/w167?link=http%3A%2F%2Ftappyplane.com%3Fsplashscreen%3DtappyWheel%26payload%3D%257B%2522data%2522%253A%2522deeplink%2520data%2522%257D%26token%3DWITBBARG%26reward%3D%255B%257B%2522amount%2522%253A%2B20%252C%2B%2522type%2522%253A%2B%2522CURRENCY%2522%252C%2B%2522id%2522%253A%2B28%257D%255D&apn=com.spilgames.tappyplane&ibi=com.spilgames.tappyplane&isi=1128336623&d=1. I test by opening this link in Safari, then tapping on "Your dynamic link" at the top of the page.

**Comment by ergunkocak** [comment]
Deeplink did not survive after app store install. If i install the app first and then click on the link it works as expected.

IOS => 12.1.1
xcode => 10.1
Firebase/Core => 5.14.0
iPhone 6S+

Also tested with IOS 12.0.1 & iPhone 6S
Also tested with wifi and cellular

Best

**Comment by dmandar** [comment]
@PeeGee85Spil We need the bundleID of the app to be the same as that configured in the Firebase console. Can you confirm the bundleID of your app (from the app store) is the same as what is you have during development?
Also, please check if you have the app preview page (https://firebase.google.com/docs/dynamic-links/link-previews) enabled and if so, is shown before you navigate to the app store.

**Comment by paulb777** [comment]
FirebaseInvites has been deprecated and we don't plan to fix this.

More at https://firebase.google.com/docs/invites/deprecation

**Comment by paulb777** [comment]
If you see a related issue with DynamicLinks, please open another issue.
